.github/actions/setup/action.yml

```diff
@@ -69,33 +69,30 @@ runs:
 
     - name: Install 'cargo-audit'
       if: ${{ contains(inputs.components, 'audit') }}
-      shell: bash
-      run: cargo install cargo-audit
+      uses: taiki-e/cache-cargo-install-action@v2
+      with:
+        tool: cargo-audit
 
     - name: Install 'cargo-hack'
       if: ${{ contains(inputs.components, 'hack') }}
-      shell: bash
-      run: cargo install cargo-hack
+      uses: taiki-e/cache-cargo-install-action@v2
+      with:
+        tool: cargo-hack
 
     - name: Install 'cargo-release'
       if: ${{ contains(inputs.components, 'release') }}
-      shell: bash
-      run: cargo install cargo-release@0.25.15
+      uses: taiki-e/cache-cargo-install-action@v2
+      with:
+        tool: cargo-release@0.25.15
 
     - name: Install 'cargo-semver-checks'
       if: ${{ contains(inputs.components, 'semver-checks') }}
-      shell: bash
-      run: cargo install cargo-semver-checks
+      uses: taiki-e/cache-cargo-install-action@v2
+      with:
+        tool: cargo-semver-checks@0.39.0
 
-    - name: Cache Cargo Dependencies
-      if: ${{ inputs.cargo-cache-key && !inputs.cargo-cache-fallback-key }}
-      uses: actions/cache@v4
+    - name: Install 'cargo-edit'
+      if: ${{ contains(inputs.components, 'edit') }}
+      uses: taiki-e/cache-cargo-install-action@v2
       with:
-        path: |
-          ~/.cargo/bin/
-          ~/.cargo/registry/index/
-          ~/.cargo/registry/cache/
-          ~/.cargo/git/db/
-          target/
-        key: ${{ runner.os }}-${{ inputs.cargo-cache-key }}-${{ hashFiles('**/Cargo.lock') }}
-        restore-keys: ${{ runner.os }}-${{ inputs.cargo-cache-key }}
```

Comment:
Oh sorry it wasn't clear, you should definitely still keep the cache for everything else!

Reply:
Will this cache be used? The cargo install action includes a cache.

Reply:
It should be used for everything you build in the workflow, right?

Reply:
Yeah, you are right – added the cache back.